.github/workflows/ci.yml

```diff
@@ -41,40 +41,110 @@ jobs:
 
     steps:
 
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
       with:
         submodules: recursive
 
-    - uses: actions/setup-python@v2
+    - uses: actions/setup-python@v3
+      with:
+        python-version: '3.9'
 
-    - name: Install
+    - name: Install Ubuntu Dependencies
       run: |
         sudo apt-get update
         sudo apt-get install git g++-9 build-essential bison flex \
           libreadline-dev gawk tcl-dev libffi-dev git graphviz xdot \
           pkg-config libboost-system-dev libboost-python-dev \
-          libboost-filesystem-dev zlib1g-dev clang-format-8 cmake
+          libboost-filesystem-dev zlib1g-dev clang-format-8 cmake \
+          nlohmann-json3-dev iverilog
```

Comment:
Aren't there googletest dependencies available in Ubuntu and homebrew that we can use directly from the package manager ?

That way we don't need the separate googletest compilation step below.

Reply:
I'm not sure if they compile to support c++ 17 string views which are needed, I can test though